docs/proposals/DNSPolicy.md

```diff
@@ -0,0 +1,390 @@
+# DNS Policy
+
+
+## Problem
+
+Gateway admins, need a way to define the DNS policy for a multi-cluster gateway in order to control how much and which traffic reaches these gateways. 
+
+**Use Cases**
+As a gateway admin, I want to be able to reduce latency for my users by distributing load geographically.
+
+As a gateway admin, I have a discount with a particular cloud provider and want to send more of my traffic to the gateways hosted in that providers infrastructure.
+
+
+
+## Goals
+
+- Allow definition of a DNS load balancing strategy to decide how traffic should be weighted across multiple gateway instances from the central control plane.
+- Allow definition of TTL and Grace Periods that control DNS TTLs and how long we give DNS caches to expire before deleting a gateway instance
+
+
+## None Goals
+
+- Allow different policies for different listeners. Although this may be something we look to support in the future, currently policy attachment does not allow for this type of targeting. This means a DNSPolicy is applied for the whole gateway currently. 
+- Define how health checks should work, this will be part of a separate proposal
+
+
+## Terms
+
+- **managed listener**: This is a listener with a host backed by a DNS zone managed by the multi-cluster gateway controller
+- **hub cluster**: control plane cluster that managed 1 or more spokes
+- **spoke cluster**: a cluster managed by the hub control plane cluster. This is where gateway are instantiated
+
+## Proposal
+
+Provide a control plane DNSPolicy API that uses the idea of direct [policy attachment](https://gateway-api.sigs.k8s.io/references/policy-attachment/#direct-policy-attachment) from gateway API that allows a load balancing strategy and TTLs to be applied to the DNS records for any managed listeners being served by the data plane instances of this gateway. 
+The DNSPolicy also covers DNS based health checks but that is not covered in this document.
+
+Below is a draft API for what we anticipate the DNSPolicy to look like
+
+```yaml 
+
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+spec:
+  targetRef: # defaults to gateway gvk and currrent namespace
+    name: gateway-name
+  health:
+   ...
+  ttl: 
+    value: 60
+    gracePeriod: 600 #(seconds to wait beyond TTL before allowing deletion of gateway) 
+  loadBalancing:
+     #one or both of geo and weighted are required 
+    weighted:
+     default: 10  #required if weighted is chosen
+     custom:
+     - attribute: cloud  #optional with both geo and weighted
+       value: AWS
+       weight: 10
+     - attribute: cloud
+       value: GCP
+       weight: 20
+    geo:
+      default: IE # required if geo is chosen
```

Comment:
It makes sense to make it more obvious that both geo and weight are not mutually exclusive and weights still apply for geo strategy. Since geo is still weighted, should setting a default for weighted be required, or will we just add a default weight if none is specified? If setting a default weight is not required, loadbalancing should probably be optional.

Reply:
Yeah still debating this one, I like having it there as it can then be used to inform other custom weights in the future. If it is hidden, then you either have to set a new one or figure out what the default we used was

Reply:
going with default is required for weight